pygyro/advection/accelerated_advection_steps.py

```diff
@@ -52,141 +52,150 @@ def poloidal_advection_step_expl( f, dt, v, rPts, qPts,
-        'double','double','double','double','double','double','double','int')
-def v_parallel_advection_eval_step( f, vPts, rPos,vMin, vMax,kts, deg,
-                        coeffs,CN0,kN0,deltaRN0,rp,CTi,kTi,deltaRTi,bound):
+                    while (endPts_k2_q[i, j] > 2*pi):
+                        endPts_k2_q[i, j] -= 2*pi
+                    while (endPts_k2_q[i, j] < 0):
+                        endPts_k2_q[i, j] += 2*pi
+                    f[i, j] = eval_spline_2d_scalar(endPts_k2_q[i, j], endPts_k2_r[i, j],
+                                                    kts1Pol, deg1Pol, kts2Pol, deg2Pol, coeffsPol, 0, 0)
+
+
+def v_parallel_advection_eval_step(f: 'float[:]', vPts: 'float[:]',
+                                   rPos: 'float', vMin: 'float', vMax: 'float',
+                                   kts: 'float[:]', deg: 'int',
+                                   coeffs: 'float[:]',
+                                   CN0: 'float', kN0: 'float', deltaRN0: 'float', rp: 'float',
+                                   CTi: 'float', kTi: 'float', deltaRTi: 'float', bound: 'int'):
+    """
+    TODO
+    """
     # Find value at the determined point
-    if (bound==0):
-        for i,v in enumerate(vPts):
-            if (v<vMin or v>vMax):
-                f[i]=f_eq(rPos,v,CN0,kN0,deltaRN0,rp,CTi,
-                                    kTi,deltaRTi)
+    if (bound == 0):
+        for i, v in enumerate(vPts):
+            if (v < vMin or v > vMax):
+                f[i] = f_eq(rPos, v, CN0, kN0, deltaRN0, rp, CTi,
+                            kTi, deltaRTi)
             else:
-                f[i]=eval_spline_1d_scalar(v,kts,deg,coeffs,0)
-    elif (bound==1):
-        for i,v in enumerate(vPts):
-            if (v<vMin or v>vMax):
-                f[i]=0.0
+                f[i] = eval_spline_1d_scalar(v, kts, deg, coeffs, 0)
+    elif (bound == 1):
+        for i, v in enumerate(vPts):
+            if (v < vMin or v > vMax):
+                f[i] = 0.0
             else:
-                f[i]=eval_spline_1d_scalar(v,kts,deg,coeffs,0)
-    elif (bound==2):
+                f[i] = eval_spline_1d_scalar(v, kts, deg, coeffs, 0)
+    elif (bound == 2):
         vDiff = vMax-vMin
-        for i,v in enumerate(vPts):
-            while (v<vMin):
-                v+=vDiff
-            while (v>vMax):
-                v-=vDiff
-            f[i]=eval_spline_1d_scalar(v,kts,deg,coeffs,0)
-
-
-@pure
-@types('int','int[:]','double[:,:,:]','double[:]','double[:]','double[:]','int','double[:]')
-@allow_negative_index('vals') # Needed for C due to pyccel issue #854
-def get_lagrange_vals(i,shifts,vals,qVals,thetaShifts,kts,deg,coeffs):
+        for i, v in enumerate(vPts):
+            while (v < vMin):
+                v += vDiff
+            while (v > vMax):
+                v -= vDiff
+            f[i] = eval_spline_1d_scalar(v, kts, deg, coeffs, 0)
+
+# @allow_negative_index('vals') # Needed for C due to pyccel issue #854
```

Comment:
I wonder if it would be better to leave this activated and change line 176 to `vals[i-s, k, j] ...`?

Reply:
Apparently, this leads to a `Aborted (core dumped)` error. I will open an issue on the `pyccel` GitHub about this.

Reply:
This is now fixed in the master branch of pyccel. It will be in the next version